core/src/validator.rs

```diff
@@ -406,6 +410,20 @@ impl Validator {
             node.sockets.retransmit_sockets[0].local_addr().unwrap()
         );
     }
+
+    fn check_vote_account(bank: &Bank, vote_account: &Pubkey) {
+        if bank.vote_accounts().get(vote_account).is_none() {
+            if let Some(found_account) = bank.get_account(vote_account) {
+                error!(
+                    "Account is not a vote account (but owned by {}): {}",
+                    found_account.owner, vote_account
+                );
+            } else {
+                error!("Vote account not found: {}:", vote_account);
+            }
+            process::exit(1);
```

Comment:
I'm not very sure why validator aborting with `process::exit(1)` around this code. Maybe `panic_hook` related?

Reply:
I’m not sure I understand what you mean?

Reply:
Well, I mean, why is there several places calling `process::exit(1)` in `validator.rs`?:

```shell
ryoqun@ubuqun:~/work/solana/solana$ git grep -B4 -A1 process::exit core/
core/src/validator.rs-            if is_x86_feature_detected!("avx") {
core/src/validator.rs-                info!("AVX detected");
core/src/validator.rs-            } else {
core/src/validator.rs-                error!("Your machine does not have AVX support, please rebuild from source on your machine");
core/src/validator.rs:                process::exit(1);
core/src/validator.rs-            }
--
core/src/validator.rs-            error!(
core/src/validator.rs-                "Delete the ledger directory to continue: {:?}",
core/src/validator.rs-                blocktree_path
core/src/validator.rs-            );
core/src/validator.rs:            process::exit(1);
core/src/validator.rs-        }
--
core/src/validator.rs-        process_options,
core/src/validator.rs-    )
core/src/validator.rs-    .unwrap_or_else(|err| {
core/src/validator.rs-        error!("Failed to load ledger: {:?}", err);
core/src/validator.rs:        process::exit(1);
core/src/validator.rs-    });
```

I think [`panic!` is preferred to than `process::exit(1)` in Rust](https://stackoverflow.com/a/39229209/10242004):

> rather use panic!() in nearly all cases

Better yet, I think because `core` rather functions as a library code, so it's not graceful to use `process::exit` in the middle of code (or `panic!` in that matter). Preferably, we should return `Result<Validator>` from `Validator::new`? Of course, some amount of rewrite will be needed... 

At least, we should comment why `process.exit` is used at the very least. The reason of `process::exit` isn't clear to me...

Reply:
> Preferably, we should return Result<Validator> from Validator::new? Of course, some amount of rewrite will be needed...

Yes exactly.  `new` functions typically never fail, so a better way I think would be to replace `Validator::new()` with a `Validator::create()` method that returns a `Result`.  

Because we don't have that nice `Validator::create()` we're using `exit(1)` in those places where really the error ought to be bubbled up to the caller (some action the user could take to resolve the issue), and reserving `panic!()` for cases where we have an internal error.

This is another clean-up item that you're very welcome to take on if you'd like!

Reply:
Thanks for clarification! Then, `process::exit` is used just by historical reasons and I'm glad that cleaning up would be welcomed. :)

Reply:
Yep, you'll see that a lot around here.   A bad code pattern gets replicated around for a while until somebody finally says "Enough!  We need to clean up this mess" and just does it :)